.github/workflows/sonarcloud.io.yml

```
@@ -0,0 +1,50 @@
+name: SonarCloud.io
+on:
+   - pull_request
+   - push
+  
+jobs:
+  build:
+    name: SonarCloud.io
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Install Java 8
+        uses: AdoptOpenJDK/install-jdk@v1
+        with:
+          version: '8'
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Maven Verify
+        run: cd app; mvn verify install -Psonarcloud -pl \!ui-react -fn                 
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Install Java 11
+        uses: AdoptOpenJDK/install-jdk@v1
+        with:
+          version: '11'
+      - name: Remove Problematic BOM project
+        run: rm -r app/extension/bom
+      - name: Remove Problematic BOM project reference
+        run: sed -i "/bom/d" app/extension/pom.xml
+      - name: Send analysis to SonarCloud
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: cd app; mvn verify -P\!non-staging -Dsonar.login=${{secrets.SONAR_TOKEN}} -Psonarcloud_send -pl \!ui-react
```

Perhaps we can simplify this a bit to:

```suggestion
        run: cd app; ./mvnw -Dsonar.login=${{secrets.SONAR_TOKEN}} sonar:sonar
```

if we declare all the `sonar.*` properties outside of the `Psonarcloud_send` profile and set the plugin/dependency on `tools.jar` based on `project.build.targetJdk` property.

---

Can we deactivate a profile and a child from the sonarcloud profile? I'm not sure how to do that.

---

From what I understand, we need to add the `tools.jar` only when running on Java 11 and targeting Java 8. So we can activate a profile that adds that dependency to the plugin dependencies based on the value of `project.build.targetJdk`.

The trick is to think of the opposite, and use profile activation, similar to how `non-staging` profile works, if a property is not defined include the modules if it's defined include it; perhaps based on the `sonar.login` property.

---

`tools.jar` is only being used when analyzing the code. We are not really going to use it outside the sonarcloud use case. Is it good to take it out of that profile then?

---

To get the `./mvnw -Dsonar.login=${{secrets.SONAR_TOKEN}} sonar:sonar` to work without specifying profiles, yeah I think it's worth it. Better have things auto-configured than rely on remembering how to invoke Maven, i.e. what profiles are needed and such.